packages/design-system/src/LoginButton.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  Avatar,
+  Box,
+  Button,
+  HStack,
+  Link,
+  Spinner,
+  Text,
+} from '@chakra-ui/react';
+import React, { useCallback } from 'react';
+
+import { MetaButton } from './MetaButton';
+
+type UserInfo = {
+  playerImage?: string;
+  playerName?: string;
+  username?: string;
+  player?: any;
+};
+
+type LoginButtonPros = {
+  connectWeb3: () => Promise<void>;
+  disconnect: () => void;
+  isConnected: boolean;
+  user: UserInfo;
+  fetching: boolean;
+};
+
+export const LoginButton: React.FC<LoginButtonPros> = ({
+  connectWeb3,
+  disconnect,
+  isConnected,
+  user,
+  fetching,
+}: LoginButtonPros) => {
+  const handleLoginClick = useCallback(async () => {
+    await connectWeb3();
+  }, [connectWeb3]);
+
+  if (isConnected) {
+    if (fetching) {
+      return <Spinner color="purple.500" size="sm" />;
+    }
+    if (!user?.player) return null;
+    return (
+      <HStack>
+        <Avatar
+          src={user.playerImage}
+          name={user.playerName}
+          as={Link}
+          href={`/player/${user.username}`}
```

Comment:
This URL may not exist on all sites where this button is used.

Maybe you could have a parameter `hrefFor` with the default value ``(user) => `/player/${user.username}`​``, then for the `href` parameter call `hrefFor(user)`.